feat(collector): add InfluxDB collector

affo opened the pull request:
# One-line summary

This PR adds the InfluxDB collector.

## Description

This PR adds the InfluxDB collector.

## Types of Changes
_What types of changes does your code introduce? Keep the ones that apply:_

- New feature (non-breaking change which adds functionality)

## Tasks
_List of tasks you will do to complete the PR_
  - [x] Implement collector
  - [x] Test in local env
  - [x] Add README

## Review
_List of tasks the reviewer must do to review the PR_
- [ ] Tests
- [ ] Documentation
- [ ] CHANGELOG

mikkeloscar commented on pkg/collector/influxdb_collector.go:
```diff
@@ -0,0 +1,150 @@
+
+func NewInfluxDBCollector(kubeClient kubernetes.Interface, hostPort string, token string, orgID string, hpa *autoscalingv2.HorizontalPodAutoscaler, config *MetricConfig, interval time.Duration) (*InfluxDBCollector, error) {
+	collector := &InfluxDBCollector{
+		kubeClient: kubeClient,
```
the `kubeClient` doesn't seem to be in use anywhere?

mikkeloscar commented on pkg/server/start.go:
```diff
@@ -88,6 +88,12 @@ func NewCommandStartAdapterServer(stopCh <-chan struct{}) *cobra.Command {
 		"whether to enable External Metrics API")
 	flags.StringVar(&o.PrometheusServer, "prometheus-server", o.PrometheusServer, ""+
 		"url of prometheus server to query")
+	flags.StringVar(&o.InfluxDBHostPort, "influxdb-hostport", o.InfluxDBHostPort, ""+
```
Maybe a better name would be "address"?

affo commented:
@mikkeloscar thank you for the quick walkthrough and for the comments.

I wanted to add the `work in progress` label to the PR, but I think I don't have permissions.

Anyways, I'll add the README section and address your comments asap.  
Once done that I'll ping you for further review. Sounds good?

Thanks again 👍

mikkeloscar commented:
You can also prefix the title with `[WIP]` then it's at least clear to me that it's not ready for review.

affo commented:
@mikkeloscar you are absolutely right  😅 Gotcha 👍

affo commented:
@mikkeloscar I guess this is ready for review.

I addressed your comments and updated the README.

I also tested it in a local env and it runs fine. I was creating an example also to help you reproducing a  working example, but I deferred it to another PR if it could be useful 👍

mikkeloscar commented:
Thanks, I'll get around to reviewing tomorrow or Friday. Please ensure you sign-off your commits so that check passes :)

affo commented:
Yeah,  gonna squash and sign.

Is the coverage step an issue? Should I add more tests?

mikkeloscar commented:
> Is the coverage step an issue? Should I add more tests?

I mean tests are always good to have :)

szuecs commented on pkg/collector/influxdb_collector.go:
```diff
@@ -0,0 +1,152 @@
+		return nil, fmt.Errorf("unknown metric type: %v", configType)
+	}
+	// Use custom InfluxDB config if defined in HPA annotation.
+	if v, found := config.Config[influxDBAddressKey]; found {
```
`found -> ok`
Also the other ones, please :)

szuecs commented on pkg/collector/influxdb_collector.go:
```diff
@@ -0,0 +1,152 @@
+				"please add metricSelector.matchLabels.%s: <...> to .yml description", influxDBQueryNameLabelKey)
+		}
+		if query, ok := config.Config[queryName]; ok {
+			// TODO(affo): validate the query.
```
should this stay or do you want to validate it before merge?

affo replied:
This should stay, that's future work 👍

szuecs replied:
ok, maybe better to add an issue and reference it from the code?

affo commented:
@szuecs addressed your comments.

@mikkeloscar added tests for collector instantiation.

Thank you for reviewing.

szuecs commented:
:+1:

mikkeloscar commented:
:+1:
